artemis-server/src/main/java/org/apache/activemq/artemis/core/server/ServiceRegistry.java

```diff
@@ -32,32 +32,12 @@
 {
```

Comment:
It's not really Immutable... the Implementation can decide on how to add / remove... 

We just need a contract to load these things... it's up to the implementation how to load / unload these.

On this case this is an application server, and they have their data already loaded somewhere. it could be immutable or not.